camera_manager.py

```diff
@@ -0,0 +1,154 @@
+import subprocess
+from time import sleep
+import cv2
+
+def get_camera_name_and_paths() -> dict[str: str]:
+    # Run the v4l2-ctl command to list available cameras
+    command = "v4l2-ctl --list-devices"
+    output = subprocess.check_output(command, shell=True, text=True)
+
+    # Create a dictionary mapping camera name to device path
+    cameras = {}
+    lines = output.strip().split('\n')
+    lines = [line.replace('\t', '') for line in lines]  # Remove tabs in output
+    lines = [line for line in lines if line != '']      # Remove empty lines
+    curr_line = 0
+    while curr_line < len(lines):
+
+        # Read the camera name
+        camera_name = lines[curr_line].split(":")[0]
+        curr_line += 1
+
+        # Read the camera file path
+        camera_path = lines[curr_line]
+        curr_line += 1
+
+        # Add camera to dictionary
+        cameras[camera_name] = camera_path
+
+        print(camera_name, camera_path)
+
+        # Skip over other device file paths
+        while curr_line < len(lines) and lines[curr_line].startswith('/dev/'):
+            curr_line += 1
+    return cameras
+
+
+
+class Camera:
+    """
+    A representation of camera that stores information like name, camera index, etc.
+    """
+    def __init__(self, camera_name: str, camera_path: str, camera_fps: int = 30):
+        self.name: str = camera_name
+        self.path: str = camera_path
+        self.fps: int = camera_fps
+        self.is_running: bool = False
+
+        """
+        Set the encoding quality for each frame that is sent to a client.
+        Although this is a list, it represents a key-value pair (WHY OPENCV?!).
+        90 is our default value, but it can range between 0-100 with 100 meaning that quality of the frame is maintained.
+        """
+        self.encoding_params: list[int] = [cv2.IMWRITE_JPEG_QUALITY, 50]
```

Comment:
The docs say 90, but the code sets it to 50.

Also, are you sure that compressing it via JPEG is a good idea? There are better formats for that - in particular, WebP. It's much more efficient while maintaining visual clarity for viewers. Python has a wealth of [libraries for this purpose](https://pypi.org/project/pillow/), though it may be a goose chase for later.

Consider changing the compression to 90 and sticking a TODO on it. :)

Reply:
yep, all fixed